crates/red_knot_python_semantic/resources/mdtest/call/methods.md

```diff
@@ -255,4 +255,81 @@ method_wrapper()
 method_wrapper(C(), C, "one too many")
 ```
 
+## `@classmethod`
+
+### Basic
+
+When a `@classmethod` attribute is accessed, it returns a bound method object, even when accessed on
+the class object itself:
+
+```py
+from __future__ import annotations
+
+class C:
+    @classmethod
+    def f(cls: type[C], x: int) -> str:
+        return "a"
+
+reveal_type(C.f)  # revealed: <bound method `f` of `Literal[C]`>
+reveal_type(C().f)  # revealed: <bound method `f` of `type[C]`>
```

Comment:
I think it might be useful to test accessing and calling a classmethod from a subclass that doesn't override it? This would ensure that we aren't placing an overly restrictive type on the `cls` arg.